toolium/driver_wrappers_pool.py

```diff
@@ -164,13 +167,18 @@ def close_drivers_and_download_videos(cls, name, test_passed=True, maintain_defa
                 # Stop driver
                 driver_wrapper.driver.quit()
                 # Download video if necessary
-                if (driver_wrapper.config.getboolean_optional('Server', 'video_enabled') or not test_passed) \
-                        and driver_wrapper.remote_node_video_enabled:
-                    driver_wrapper.utils.download_remote_video(driver_wrapper.remote_node, driver_wrapper.session_id,
-                                                               video_name.format(name, driver_index))
-            except Exception:
-                # Capture exceptions to avoid errors in teardown method due to session timeouts
-                pass
+                if not test_passed and driver_wrapper.config.getboolean_optional('Server', 'video_enabled', False):
+                    if driver_wrapper.remote_node_video_enabled:
+                        # Download video using Selenium
+                        driver_wrapper.utils.download_remote_video(driver_wrapper.remote_node, driver_wrapper.session_id,
+                                                                   video_name.format(name, driver_index))
+                    elif driver_wrapper.config.getboolean_optional('Server', 'selenoid', False):
+                        # Download video and log session using Selenoid
+                        s = Selenoid(driver_wrapper, videos_dir=cls.videos_directory, logs_dir=cls.logs_directory)
+                        s.download_video_if_the_scenario_fails(video_name.format(name, driver_index))
+                        s.download_session_log()
```

Comment:
download_session_log should be out of close_drivers_and_download_videos(). It must be possible to download logs without downloading video

Reply:
Fixed in 2bcebd5